engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1773,6 +1773,139 @@ static int SetSound(lua_State* L) {
+ * @param table [type:table] A table containing info about how to create the buffer. Supported entries:
+ *
+ * * `buffer`
+ * : [type:buffer] the buffer resource to bind to this resource
+ *
+ * * `transfer_ownership`
+ * : [type:boolean] optional flag to determine wether or not the resource should take over ownership of the buffer object (default true)
+ *
+ * @return path [type:hash] Returns the buffer resource path
+ *
+ * @examples
+ * Create a buffer object and bind it to a buffer resource
+ *
+ * ```lua
+ * function init(self)
+ *     local size = 1
+ *     local positions = {
+ *         -- triangle 1
+ *          size,  size, 0,
+ *         -size, -size, 0,
+ *          size, -size, 0,
+ *         -- triangle 2
+ *          size, size,  0,
+ *         -size,  size, 0,
+ *         -size, -size, 0,
+ *     }
+ *
+ *     local buffer_handle = buffer.create(#positions, {
+ *         {
+ *             name  = hash("position"),
+ *             type  = buffer.VALUE_TYPE_FLOAT32,
+ *             count = 3
+ *         }
+ *     })
+ *
+ *     local stream = buffer.get_stream(buffer_handle, hash("position"))
+ *
+ *     -- transfer vertex data to buffer
+ *     for k=1,#positions do
+ *         stream[k] = positions[k]
+ *     end
+ *
+ *     local my_buffer = resource.create_buffer("/my_buffer.bufferc", { buffer = buffer_handle })
+ *     go.set("/go#mesh", "vertices", my_buffer)
+ * end
+ * ```
+ */
+static int CreateBuffer(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+
+    const char* path_str     = luaL_checkstring(L, 1);
+    const char* resource_ext = ".bufferc";
+
+    dmhash_t canonical_path_hash = 0;
+    PreCreateResource(L, path_str, resource_ext, &canonical_path_hash);
+
+    luaL_checktype(L, 2, LUA_TTABLE);
+    lua_pushvalue(L, 2);
+
+    lua_getfield(L, -1, "buffer");
+    dmScript::LuaHBuffer* lua_buffer = dmScript::CheckBuffer(L, -1);
+    lua_pop(L, 1); // "buffer"
+
+    bool transfer_ownership = CheckFieldValue<bool>(L, -1, "transfer_ownership", true);
+
+    lua_pop(L, 1); // args table
+
+    dmGameObject::HInstance sender_instance = dmScript::CheckGOInstance(L);
+    dmGameObject::HCollection collection    = dmGameObject::GetCollection(sender_instance);
+
+    // JG: We have to do this awkwardness because the create functions for the resource
+    //     dmDDF::LoadMessage won't accept an empty dmBufferDDF::BufferDesc structure
+    //     and the buffer res won't accept a bufferdesc a stream that doesn't have data
+    int dummy_value_ptr = 0;
+    dmBufferDDF::StreamDesc buffer_stream_dummy_ddf = {};
+    buffer_stream_dummy_ddf.m_ValueType  = dmBufferDDF::VALUE_TYPE_INT8;
+    buffer_stream_dummy_ddf.m_ValueCount = 1;
+    buffer_stream_dummy_ddf.m_I.m_Data   = &dummy_value_ptr;
+    buffer_stream_dummy_ddf.m_I.m_Count  = 1;
+
+    dmBufferDDF::BufferDesc buffer_desc_dummy_ddf = {};
+    buffer_desc_dummy_ddf.m_Streams.m_Data = &buffer_stream_dummy_ddf;
+    buffer_desc_dummy_ddf.m_Streams.m_Count = 1;
```

Comment:
What's our alternatives here? We could add a m_Message pointer to the create/preload functions to the resource api, but do we really want that..